app/src/main/java/com/willowtree/vocable/presets/PresetsFragment.kt

```diff
@@ -275,24 +274,24 @@ class PresetsFragment : BaseFragment() {
     }
 
     inner class CategoriesPagerAdapter(fm: FragmentManager) :
-        FragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
+        VocableFragmentStateAdapter(fm, viewLifecycleOwner.lifecycle) {
 
         private val categories = mutableListOf<Category>()
 
-        var numPages = 0
+        override fun setItems(items: List<Any>) {
+            super.setItems(items)
 
-        fun setCategories(categories: List<Category>) {
-            with(this.categories) {
+            with(categories) {
                 clear()
-                addAll(categories)
+                items.forEach {
```

Comment:
Think you can condense this down to `addAll( items.filter { it is Category } )`

Reply:
Also, do you expect some of these types to _not_ be `Category`?

Reply:
So `items` could be either a list of `Category` or a list of `Phrase`. So I was having trouble casting the `items` list to the right type in each of the subclasses. Would making it generic work for that? So I could have a `VocableFragmentStateAdapter<Category>` and `setItems` would be `setItems(items: List<Category>)`?

Reply:
It should yeah

You could do `VocableFragmentStateAdapter<T>(fm: FragmentManager, lifecycle: Lifecycle)`
and then change `setItems` to `open fun setItems(items: List<T>)`

Reply:
Cool, just updated the adapters